content/Watsonx/WatsonxData/106.md

```diff
@@ -0,0 +1,202 @@
+
+  ```bash
+    docker exec ibm-lh-postgres printenv | grep POSTGRES_PASSWORD | sed 's/.*=//'
+  ```
+
+9. Repeat Steps 2-6 from above to add the PostgreSQL database to watsonx.data. Use the following information:
+
+            • Database type:    PostgreSQL(it can be found under Other Databases)
+            • Database name:    gosales
+            • Display name:     PostgreSQLDB
+            • Hostname:         192.168.252.2
+            • Port:             5432
+            • Username:         admin
+            • Password:         ***The password generated in the previous step***
+            • Catalog name:     pgcatalog
+
+
+With both the Db2 and PostgreSQL databases added, the topology should look like the image below.
+
+  ![](./images/106/add-database-output-pg.png)
+
+10. Select the **Data manager** icon from the left-side menu.
+
+11. Expand both the db2catalog and pgcatalog catalogs.
+
+  ![](./images/106/data-manager-view.png)
+
+  Schema and table information are shown for both of these databases. Both include an identical **gosalesdw** schema with tables containing sales data for the fictional Great Outdoors company. Remember that there is also a copy of this same data in object storage, managed by the **hive_data** catalog.
+
+  If you do not see any tables associated with the gosalesdw schema in either of the catalogs, you can refresh the schema.
+
+
+  ![](./images/106/data-manager-refresh.png)
+
+  > **Note:** If for some reason the list of tables does not refresh, or if you see a message appear in the upper-right corner saying ***“An error occurred while loading the tables in gosalesdw”***, similar to the image below, you can still proceed. The table information is in the catalog and queries that reference these tables will still work. Please proceed with the rest of the lab.
+
+  ![](./images/106/data-manager-refresh-error.png)
+
+
+  You now have copies of the Great Outdoors company sales tables in object storage (hive_data), Db2 (db2catalog), and PostgreSQL (pgcatalog). This isn’t something you’d have in a real-world scenario (after all, the benefit of being able to federate access to multiple data sources is to avoid data duplication). However, it’s being done here for the purposes of highlighting Presto’s federation capabilities.
+
+  You will now see how you can run a federated query that combines data from all three of these data sources.
+
+12. Select the **Query workspace** icon from the left-side menu.
+
+13. Copy and paste the query below into the **SQL worksheet**. Click Run on **presto-01**.
+
+  This sample query could be used by the fictional business to determine which purchasing method is associated with the largest orders. The query accesses five tables, one of which is in PostgreSQL, two are in Db2, and two are in watsonx.data’s object storage.
+
+  ```bash
```

Comment:
Is there a way we could indent this code so it appears more like it does in the original lab guide?

Reply:
tried ..looks similar now